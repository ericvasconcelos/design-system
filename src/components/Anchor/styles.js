import styled from 'styled-components'
import { getButtonStyle } from 'utils/getButtonStyle'

export const Wrapper = styled.a`
  ${getButtonStyle}
`
