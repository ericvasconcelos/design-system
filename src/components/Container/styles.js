import styled from 'styled-components'

export const Wrapper = styled.div`
  margin: auto;
  width: ${({ theme }) => theme.grid.smMobile[1]};

  @media (min-width: 576px) {
    width: ${({ theme }) => theme.grid.mobile[1]};
  }

  @media (min-width: 768px) {
    width: ${({ theme }) => theme.grid.smTablet[1]};
  }

  @media (min-width: 992px) {
    width: ${({ theme }) => theme.grid.tablet[1]};
  }

  @media (min-width: 1200px) {
    width: ${({ theme }) => theme.grid.laptop[1]};
  }

  @media (min-width: 1400px) {
    width: ${({ theme }) => theme.grid.desktop[1]};
  }
`
