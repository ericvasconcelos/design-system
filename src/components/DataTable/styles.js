import { Icon } from 'components/Icon'
import { Loader as LoaderComponent } from 'components/Loader'
import { Text as TextComponent } from 'components/Text'
import styled from 'styled-components'
import { getColorFromName } from 'utils/getColorFromName'
import { getTypography } from 'utils/getTypography'

export const WrapperTable = styled.div`
  @media (max-width: 1024px) {
    overflow-x: auto;
  }
`

export const Loader = styled(LoaderComponent)`
  margin: auto;
  margin-top: unset;
  height: 100%;
`

export const Table = styled.table`
  border-collapse: separate;
  width: 100%;
  margin: auto;
  border-spacing: ${({ theme }) => `0 ${theme.gutters.sm}`};

  td {
    ${({ theme }) =>
    getTypography({
      theme,
      styleType: 'body2'
    })}
  }

  td:first-child,
  th:first-child {
    border-radius: 8px 0 0 8px;
  }

  td:last-child,
  th:last-child {
    border-radius: 0 8px 8px 0;
  }
`

export const Header = styled.th`
  padding: ${({ theme }) => theme.padding.md};
  position: relative;
  cursor: ${({ sortable }) => (sortable ? 'pointer' : 'auto')};
`

export const HeaderContent = styled.div`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.gutters.sm};
  white-space: nowrap;
  text-align: left;
  ${({ theme }) =>
    getTypography({
      theme,
      styleType: 'body2Medium',
      styleColor: 'darkBlueCash500'
    })}
`

export const IconSort = styled(Icon)`
  width: 16px;
  height: 16px;
  mask-size: 16px;
  background: ${({ theme }) => theme.pallete.darkBlueCash[500]};
`

export const Column = styled.th`
  padding: ${({ theme }) => theme.padding.md};
  text-align: left;
  border: 1px solid transparent;
  &:focus {
    border: 1px solid ${({ theme }) => theme.pallete.lightBlueCash[200]};
    box-shadow: 0 0 0 4px
      ${({ theme }) => getColorFromName({ theme, styleColor: 'lightBlueCash300-24' })};
  }
  ${({ theme }) =>
    getTypography({
      theme,
      styleType: 'body2',
      styleColor: 'darkBlueCash500'
    })}
`

export const ClickableColumn = styled(Column)`
  cursor: pointer;
  padding: 0;
  vertical-align: middle;
  line-height: 0;
  max-width: 80px;

  
  
  `

export const Row = styled.tr`
  border-bottom: 1px solid rgba(0, 0, 0, 0.08);
  display: ${({ collapse }) => (collapse ? 'none' : 'table-row')};
  background-color: ${({ theme }) => theme.pallete.greyCash[100]};
  border-top-left-radius: 10px;
  
  

`

export const Pagination = styled.div`
  text-align: center;
`

export const ShowMoreButton = styled.button`
  display: flex;
  justify-content: center;
  &:focus {
    border: 1px solid ${({ theme }) => theme.pallete.lightBlueCash[200]};
    box-shadow: 0 0 0 4px
      ${({ theme }) => getColorFromName({ theme, styleColor: 'lightBlueCash300-24' })};
  }
  ${({ theme }) =>
    getTypography({
      theme,
      styleType: 'body2Medium',
      styleColor: 'lightBlueCash300'
    })}
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
  background-color: ${({ theme }) => theme.pallete.greyCash[100]};
  width: 100%;
  border-radius: 8px;
  padding: ${({ theme }) => theme.padding.lg};
`

export const Text = styled(TextComponent)`
  display: flex;
  justify-content: center;
  ${({ theme }) =>
    getTypography({
      theme,
      styleType: 'body2',
      styleColor: 'darkBlueCash500'
    })}
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'auto')};
  background-color: ${({ theme }) => theme.pallete.greyCash[100]};
  width: 100%;
  border-radius: 8px;
  padding: ${({ theme }) => theme.padding.lg};
`

export const ButtonIcon = styled.button`
  cursor: pointer;
`
