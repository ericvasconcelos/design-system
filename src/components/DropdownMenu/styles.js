import styled from 'styled-components'
import { getColorFromName } from 'utils/getColorFromName'

export const Wrapper = styled.ul`
  background-color: ${({ theme }) => theme.pallete.greyCash[100]};
  border: 1px solid ${({ theme }) => theme.pallete.greyCash[300]};
  padding: ${({ theme }) => theme.padding.sm};
  border-radius: 1rem;
  box-shadow: 0px 8px 16px 0px
    ${({ theme }) => getColorFromName({ theme, styleColor: 'greyCash500-08' })};
  list-style: none;
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.gutters.sm};
  align-items: stretch;
`
