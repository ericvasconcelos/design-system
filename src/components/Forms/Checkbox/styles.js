import styled from 'styled-components'
import { getColorFromName } from 'utils/getColorFromName'

export const Label = styled.label`
  display: flex;
  align-items: center;
`

export const StyledInput = styled.input`
  appearance: none;

  background-color: ${({ theme, disabled }) =>
    disabled
      ? getColorFromName({ theme, styleColor: 'greyCash400-08' })
      : theme.pallete.greyCash[100]};
  border: 2px solid
    ${({ theme, disabled }) =>
    disabled
      ? getColorFromName({ theme, styleColor: 'greyCash400-08' })
      : theme.pallete.greyCash[300]};
  border-radius: ${({ theme }) => theme.border.radius.xs};
  display: inline-block;
  position: relative;
  padding: 10px;
  margin-right: ${({ theme }) => theme.gutters.md};

  &[indeterminate] {
    background-color: ${({ theme }) => theme.pallete.success[400]};
    border-color: ${({ theme }) => theme.pallete.success[400]};

    &:after {
      content: '';
      width: 8px;
      height: 2px;
      background-color: ${({ theme }) => theme.pallete.greyCash[100]};
      position: absolute;
      top: 9px;
      left: 6px;
      border-radius: ${({ theme }) => theme.border.radius.xs};
    }
  }

  &:not([disabled]):hover,
  &:not([disabled]):focus {
    border-color: ${({ theme }) => theme.pallete.lightBlueCash[300]};
    box-shadow: 0 0 0 4px
      ${({ theme }) => getColorFromName({ theme, styleColor: 'lightBlueCash300-24' })};
  }

  &[checked] {
    ${({ disabled, theme }) =>
    !disabled &&
      `
        background-color: ${theme.pallete.success[400]};
        border-color: ${theme.pallete.success[400]};
      `}

    &:after {
      content: ' ';
      mask-image: url(${({ theme }) => theme.icons.checkBold});
      mask-size: contain;
      background-color: ${({ theme, disabled }) =>
    disabled
      ? getColorFromName({ theme, styleColor: 'greyCash400-32' })
      : theme.pallete.greyCash[100]};
      width: 8px;
      height: 6px;
      position: absolute;
      top: 7px;
      left: 6px;
    }
  }
`
