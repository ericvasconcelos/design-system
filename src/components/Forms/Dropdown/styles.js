import { Button as BaseButton } from 'components/Button'
import { Icon } from 'components/Icon'
import styled from 'styled-components'
import { getColorFromName } from 'utils/getColorFromName'
import { getTypography } from 'utils/getTypography'

import { BaseIconWrapper, BaseInput, BaseInputWrapper, BaseLabel } from '../commonStyles'

export const IconWrapper = styled(BaseIconWrapper)`
  grid-area: icon;
`

export const OptionIcon = styled(Icon)``

export const CustomSelect = styled(BaseInputWrapper)`
  display: grid;
  grid-template: 'label icon' 1.0625rem 'input icon' 1.375rem / minmax(0, 1fr) 1.5rem;
  column-gap: 1rem;
  row-gap: 0.1875rem;
  position: relative;
  white-space: nowrap;
`

export const CustomSelectCheckbox = styled(CustomSelect)`
  cursor: pointer;
`

export const Label = styled(BaseLabel)`
  grid-area: label;
`

export const DisplaySelection = styled(BaseInput)`
  grid-area: input;
  overflow: hidden;
  text-overflow: ellipsis;
  color: ${({ value, theme }) =>
    value === '' ? theme.pallete.darkBlueCash[200] : theme.pallete.darkBlueCash[500]};
`

export const OptionsBox = styled.ul`
  background-color: ${({ theme }) => theme.pallete.greyCash[100]};
  border-radius: 0.5rem;
  border: 1px solid;
  border-color: ${({ theme }) => theme.pallete.greyCash[300]};
  height: calc(
    ${({ numberOptions, optionSize }) =>
    `${optionSize === 'large' ? '3.5rem' : '2.5rem'} * ${
      numberOptions > 5 ? 5 : numberOptions
    } + 1rem`}
  );
  width: 100%;
  box-sizing: border-box;
  z-index: 10;
  overflow-y: auto;
  display: ${({ active }) => (active ? 'block' : 'none')};
  padding: ${({ theme }) => theme.padding.sm};
`

const BaseOption = styled.li`
  display: flex;
  align-items: center;
  ${({ theme }) => getTypography({ theme, styleType: 'input', styleColor: 'darkBlueCash500' })};
  padding: ${({ theme }) => theme.padding.md};
  gap: ${({ theme }) => theme.padding.sm};
  line-height: 1;
  border-radius: 0.25rem;
  pointer-events: auto;
  height: ${({ optionSize }) => (optionSize === 'large' ? '3.5rem' : '2.5rem')};
`

export const Option = styled(BaseOption)`
  cursor: pointer;
  background-color: ${({ theme, hovered }) =>
    hovered ? getColorFromName({ theme, styleColor: 'lightBlueCash300-24' }) : 'unset'};

  :hover {
    background-color: ${({ theme }) =>
    getColorFromName({ theme, styleColor: 'lightBlueCash100' })};
    color: ${({ theme }) => theme.pallete.lightBlueCash[300]};

    ${OptionIcon} {
      background-color: ${({ theme }) => theme.pallete.lightBlueCash[100]};
    }
  }

  :focus {
    background-color: ${({ theme }) =>
    getColorFromName({ theme, styleColor: 'lightBlueCash100' })};
    color: ${({ theme }) => theme.pallete.lightBlueCash[300]};

    ${OptionIcon} {
      background-color: ${({ theme }) => theme.pallete.lightBlueCash[300]};
    }
  }
`

export const OptionCheckbox = styled(BaseOption)`
  label,
  input {
    cursor: pointer;
  }
`

export const HiddenSelect = styled.select`
  visibility: hidden;
  position: absolute;
  top: 0;
  left: 0;
`

export const CheckboxOptionsWrapper = styled.div`
  display: ${({ active }) => (active ? 'block' : 'none')};
  background-color: ${({ theme }) => theme.pallete.greyCash[100]};
  position: relative;
  height: calc(
    ${({ numberOptions, optionSize }) =>
    `${optionSize === 'large' ? '3.5rem' : '2.5rem'} * ${
      numberOptions > 4 ? 4 : numberOptions
    } + 1rem`}
  );
  border-radius: 0.5rem;
  border: 1px solid;
  border-color: ${({ theme }) => theme.pallete.greyCash[300]};
`

export const CheckboxOptionsBox = styled.ul`
  height: calc(
    ${({ numberOptions, optionSize }) =>
    `${optionSize === 'large' ? '3.5rem' : '2.5rem'} * ${
      numberOptions > 3 ? 3 : numberOptions
    } + 1rem`}
  );
  border-radius: 0.5rem;
  width: 100%;
  box-sizing: border-box;
  z-index: 10;
  overflow-y: auto;
  padding: ${({ theme }) => theme.padding.sm};
`

export const ClearOptions = styled.div`
  display: inline-flex;
  width: 100%;
  justify-content: center;
  align-items: center;
  border-top: 1px solid ${({ theme }) => theme.pallete.greyCash[300]};
  border-radius: 0rem 0rem 0.5rem 0.5rem;
  ${({ theme }) => getTypography({ theme, styleType: 'body2', styleColor: 'lightBlueCash300' })};
  height: ${({ optionSize }) => (optionSize === 'large' ? '3.5rem' : '2.5rem')};
`

export const Button = styled(BaseButton)`
  text-transform: unset;
  height: 100%;
  font-weight: 500 !important;
  width: auto;
`
