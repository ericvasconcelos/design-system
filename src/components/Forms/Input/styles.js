import styled from 'styled-components'
import { getTypography } from 'utils/getTypography'

import {
  BaseIconWrapper,
  BaseInput,
  BaseInputWrapper,
  BaseLabel,
  BaseWrapper
} from '../commonStyles'

const getLabelState = (active, hasValue, autoFilled, readOnly, disabled) => {
  const isLabelPlaceholderAlike = !(hasValue || active || autoFilled || readOnly || disabled)
  return isLabelPlaceholderAlike
}

const getLabelBottom = ({ active, hasValue, autoFilled, readOnly, disabled }) => {
  let bottom = 0
  if (getLabelState(active, hasValue, autoFilled, readOnly, disabled)) {
    bottom = '-0.5625rem'
  }
  return bottom
}

const getLabelTypography = ({ theme, active, hasValue, autoFilled, readOnly, disabled }) => {
  if (getLabelState(active, hasValue, autoFilled, readOnly, disabled)) {
    return getTypography({ theme, styleType: 'body1', styleColor: 'darkBlueCash200' })
  }
}
export const Wrapper = styled(BaseWrapper)``

export const InputWrapper = styled(BaseInputWrapper)`
  display: grid;
  grid-template: 'label icon' 1.0625rem 'input icon' 1.375rem / minmax(0, 1fr) 1.5rem;
  column-gap: 1rem;
  row-gap: 0.1875rem;
  position: relative;
`

export const Label = styled(BaseLabel)`
  position: relative;
  bottom: ${getLabelBottom};
  ${getLabelTypography}
  grid-area: label;
`

export const CustomInput = styled(BaseInput)`
  grid-area: input;
`

export const IconWrapper = styled(BaseIconWrapper)`
  grid-area: icon;
`
