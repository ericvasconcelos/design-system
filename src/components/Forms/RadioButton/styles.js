
import styled from 'styled-components'
import { getColorFromName } from 'utils/getColorFromName'

export const Label = styled.label`
  display: flex;
  align-items: center;
`

export const StyledInput = styled.input`
  appearance: none;
  background-color: ${({ theme, disabled }) =>
    disabled
      ? getColorFromName({ theme, styleColor: 'greyCash500-08' })
      : theme.pallete.greyCash[100]};
  border: 2px solid
    ${({ theme, disabled }) =>
    disabled
      ? getColorFromName({ theme, styleColor: 'greyCash500-08' })
      : theme.pallete.greyCash[300]};
  border-radius: ${({ theme }) => theme.border.radius.rounded};
  display: inline-block;
  position: relative;
  padding: 6px;
  margin: ${({ theme }) => `${theme.gutters.xs} 12px ${theme.gutters.xs} 0`};

  &:not([disabled]):hover,
  &:not([disabled]):focus {
    border-color: ${({ theme }) => theme.pallete.lightBlueCash[300]};
    box-shadow: 0 0 0 4px
      ${({ theme }) => getColorFromName({ theme, styleColor: 'lightBlueCash300-24' })};
  }

  &[checked] {
    padding: 3px;
    border-width: 5px;
    border-color: ${({ disabled, theme }) =>
    disabled
      ? getColorFromName({ theme, styleColor: 'greyCash500-08' })
      : theme.pallete.success[400]};
    background-color: ${({ disabled, theme }) =>
    disabled
      ? getColorFromName({ theme, styleColor: 'greyCash400-32' })
      : theme.pallete.greyCash[100]};
  }
`
