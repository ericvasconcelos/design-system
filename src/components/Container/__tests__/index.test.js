import React from 'react'

import { theme } from 'styles/theme'
import { styledRender } from 'utils/styledRender'

import { Container } from '../index.js'

describe('should render the component correctly when', () => {
  it('has basic usage', () => {
    const expectedCss = `
        margin: auto;
        width: ${theme.grid.smMobile[1]};
      }
      @media (min-width: 360px) {
        width: ${theme.grid.mobile[1]};
      }
      @media (min-width: 768px) {
        width: ${theme.grid.smTablet[1]};
      }
      @media (min-width: 1024px) {
        width: ${theme.grid.tablet[1]};
      }
      @media (min-width: 1280px) {
        width: ${theme.grid.laptop[1]};
      }
      @media (min-width: 1920px) {
        width: ${theme.grid.desktop[1]};
      }
      {}}}
    `
    const { getByTestId } = styledRender(
      <Container>
        <div data-testid="content">Consectetur do ipsum et voluptate ad nisi nisi sint.</div>
      </Container>
    )
    const component = getByTestId('Container')
    expect(component).toHaveStyle(expectedCss)
    expect(expect(getByTestId('content')).toBeInTheDocument())
    expect(component).toMatchSnapshot()
  })
})
