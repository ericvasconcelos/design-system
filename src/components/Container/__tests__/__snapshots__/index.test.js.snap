// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly when has basic usage 1`] = `
.c0 {
  margin: auto;
  width: 100%;
}

@media (min-width:576px) {
  .c0 {
    width: 540px;
  }
}

@media (min-width:768px) {
  .c0 {
    width: 720px;
  }
}

@media (min-width:992px) {
  .c0 {
    width: 960px;
  }
}

@media (min-width:1200px) {
  .c0 {
    width: 1140px;
  }
}

@media (min-width:1400px) {
  .c0 {
    width: 1320px;
  }
}

<div
  class="c0"
  data-testid="Container"
>
  <div
    data-testid="content"
  >
    Consectetur do ipsum et voluptate ad nisi nisi sint.
  </div>
</div>
`;
